pkg/steps/multi_stage/gen.go

```diff
@@ -163,6 +163,14 @@ func (s *multiStageTestStep) generatePods(
 				pod.Spec.DNSPolicy = coreapi.DNSNone
 			}
 		}
+		if step.NodeArchitecture != nil {
+			if pod.Spec.NodeSelector == nil {
```

Comment:
This seems smelly. How the NodeSelector can be nil or not? We are creating the pod. It seems that we are using the pod from [steps/pod.go#L248-L250](https://github.com/deepsm007/ci-tools/blob/c963d9b5984fbdfb2652e67d25ce336c0f4535c9/pkg/steps/pod.go#L248-L250).

Reply:
You need to handle the nodeSelector in https://github.com/openshift/ci-tools/blob/master/pkg/steps/multi_stage/gen.go#L137-L139 through the generator options.

Reply:
Addressed in the commit